File: core/src/nl/t64/game/rpg/screens/world/entity/events/LoadEntityEvent.java
package nl.t64.game.rpg.screens.world.entity.events;

import com.badlogic.gdx.math.Vector2;
import nl.t64.game.rpg.screens.world.entity.Direction;
import nl.t64.game.rpg.screens.world.entity.EntityState;


public class LoadEntityEvent implements Event {

    public final EntityState state;
    public final Direction direction;
    public final Vector2 position;
    public final String conversationOrBattleId;

    public LoadEntityEvent(Vector2 position) {
        this(null, null, position, null);
    }

    public LoadEntityEvent(Direction direction, Vector2 position) {
        this(null, direction, position, null);
    }

    public LoadEntityEvent(EntityState state, Vector2 position) {
        this(state, null, position, null);
    }

    public LoadEntityEvent(EntityState state, Direction direction, Vector2 position) {
        this(state, direction, position, null);
    }

    public LoadEntityEvent(EntityState state, Direction direction, Vector2 position, String conversationOrBattleId) {
        this.state = state;
        this.direction = direction;
        this.position = position;
        this.conversationOrBattleId = conversationOrBattleId;
    }

}

File: core/src/nl/t64/game/rpg/screens/world/entity/events/LoadEntityEvent.kt
package nl.t64.game.rpg.screens.world.entity.events

import com.badlogic.gdx.math.Vector2
import nl.t64.game.rpg.screens.world.entity.Direction
import nl.t64.game.rpg.screens.world.entity.EntityState


class LoadEntityEvent : Event {

    val state: EntityState?
    val direction: Direction?
    val position: Vector2
    val conversationOrBattleId: String?

    constructor(position: Vector2) {
        this.state = null
        this.direction = null
        this.position = position
        this.conversationOrBattleId = null
    }

    constructor(direction: Direction, position: Vector2) {
        this.state = null
        this.direction = direction
        this.position = position
        this.conversationOrBattleId = null
    }

    constructor(state: EntityState, position: Vector2) {
        this.state = state
        this.direction = null
        this.position = position
        this.conversationOrBattleId = null
    }

    constructor(state: EntityState, direction: Direction, position: Vector2) {
        this.state = state
        this.direction = direction
        this.position = position
        this.conversationOrBattleId = null
    }

    constructor(state: EntityState, direction: Direction, position: Vector2, conversationOrBattleId: String) {
        this.state = state
        this.direction = direction
        this.position = position
        this.conversationOrBattleId = conversationOrBattleId
    }

}
